package-files/systemd/infra.service

```diff
@@ -0,0 +1,14 @@
+[Unit]
+Description=Infra SSH Connector
+Documentation=https://docs.infrahq.com
+Wants=network-online.service
+After=network-online.service
+ConditionPathExists=/etc/infra/connector.yaml
+
+[Service]
+Type=simple
+ExecStart=/usr/local/sbin/infra connector -f /etc/infra/connector.yaml
```

Comment:
This service should ideally run as `infra:infra` but we'll need to figure out what if it needs privilege and what those privileges are

Reply:
The main thing is calling `useradd`, `userdel`, and `pkill` (https://github.com/infrahq/infra/blob/main/docs/dev/ssh/README.md).  All users should be able to read `/etc/passwd`, and the rest is API calls to infra.

If we don't run this as root then we'll need to call `sudo` and make sure the infra user is granted password-less sudo access to those commands.

Are there other options that allow us to run it as non-root ?

Reply:
I see `CAP_KILL` for pkill I guess, but I'm not seeing anything for adding/removing users. I guess it makes sense to keep those only for root, since it effectively lets you create other root users.

Reply:
add/remove user are file operations, not syscalls, so there won't be caps for those

Reply:
I've tried this with User/Group as infra and sudoers allowing `/usr/sbin/useradd`, `/usr/sbin/userdel`, and `/usr/bin/pkill` without any issues though it probably didn't exercise all code paths